src/Runtime/ExecutionSession.cpp

```diff
@@ -32,6 +34,15 @@ const std::string ExecutionSession::_outputSignatureName = "omOutputSignature";
 
 ExecutionSession::ExecutionSession(
     std::string sharedLibPath, bool defaultEntryPoint) {
+
+  // For some reason, the call to get the permanent library requires a path, and
+  // when none is provided, it still does not look in the current directory. To
+  // remediate this situation, if no path is given, we add a "./" to the path to
+  // force looking in the current directory.
+  if (llvm::sys::path::parent_path(sharedLibPath).empty()) {
+    sharedLibPath = "./" + sharedLibPath;
+    std::cout << "hi alex, added local " << sharedLibPath << std::endl;
+  }
   _sharedLibraryHandle =
```

Comment:
I don't think the two are directly comparable. `omCompileFromFile` is a compile time function whereas `ExecutionSession` is a runtime function. So it's not a problem that they might have different search behavior. But the main point is that we shouldn't be doing things under the cover to change the search behavior not expected by the caller.

Reply:
For a compile section I would expect to search for the file to compile in the current directory. And if not found emit an error. I don't think the compile interface need to change.

Reply:
> For a compile section I would expect to search for the file to compile in the current directory. And if not found emit an error. I don't think the compile interface need to change.

@etiotto so would't we expect the same for loading the .so file when starting an execution session? 

Or maybe the right way to do it is that If the compiler interface returns the .so with a fully qualified path, so that it flows naturally from the compiler to the execution session, and I don't need to add the "special casing" in the loading of the file to artificially look in the "./" too, since that does not sit well with @gongsu832 . What do you both think about this?

Reply:
@AlexandreEichenberger 

> ```c++
>   rc = onnx_mlir::omCompileFromFile("add.onnx", "add-cppinterface",
>       onnx_mlir::EmitLib, &compiledFilename, &errorMessage);
> ```
> succeed without the ./ because LLVM seems to search the current dir, so this lead to a strange situation where the path for the compilation could skip ./ but the execution session did not.

What file(s) are you referring to when you say "succeed without the ./ because LLVM seems to search the current dir"?

As mentioned before, compilation search by LLVM has it own rules, and runtime search is not done by LLVM but rather ld.so, the dynamic loader, which has its own rules. So I'm not sure why it is confusing when the behaviors are different.

Reply:
Agree with @gongsu832. The compilation command:
```
gcc -shared -o libfoo.so foo.c 
```
searches for the input file `foo.c` in the current directory (that is without having to specify ./foo.c).

Conversely to run the executable one needs to provide the relative path (that is prepend the exec with a dot: ./bar). 
Also the executable depends on a shared library which will only be found if the directory containing the shared library is in the LD_LIBRARY_PATH). 

In summary it make sense to me that the compilation and execution environment do not need to behave in the same way.

Reply:
The issue is that we don't have the ability to add a `-L` option programmatically at this point.